sdcm/read_operation_thread.py

```diff
@@ -81,25 +85,64 @@ def run_fullscan(self, db_node: BaseNode):  # pylint: disable=too-many-locals
                         result.fetch_next_page()
                         if read_pages > 0:
                             pages += 1
-                    fs_event.message = "full scan ended successfully"
+                    operation_event.message = f"{read_operation_event.__name__} operation ended successfully"
                 except Exception as exc:  # pylint: disable=broad-except
                     msg = str(exc)
                     msg = f"{msg} while running Nemesis: {db_node.running_nemesis}" if db_node.running_nemesis else msg
-                    fs_event.message = msg
+                    operation_event.message = msg
 
                     if db_node.running_nemesis or any(s in msg.lower() for s in ERROR_SUBSTRINGS):
-                        fs_event.severity = Severity.WARNING
+                        operation_event.severity = Severity.WARNING
                     else:
-                        fs_event.severity = Severity.ERROR
+                        operation_event.severity = Severity.ERROR
 
-    def run(self):
+    def run_for_a_duration(self, read_operation_event):
         end_time = time.time() + self.duration
         while time.time() < end_time and not self.termination_event.is_set():
-            self.run_fullscan(db_node=random.choice(self.db_cluster.nodes))
+            self.run_read_operation(read_operation_event=read_operation_event)
             time.sleep(self.interval)
 
+    @abstractmethod
+    def run(self):
+        ...
+
     def start(self):
         self._thread.start()
 
     def join(self, timeout=None):
         return self._thread.join(timeout)
+
+
+class FullScanThread(ReadOperationThread):
+
+    def __init__(self, **kwargs):
+        super().__init__(**kwargs)
+        self.query_options = [self.basic_query, self.basic_query + self.bypass_cache]
+
+    def randomly_form_cql_statement(self) -> Optional[str]:
+        cmd = random.choice(self.query_options).format(self.ks_cf)
+        return self.randomly_add_timeout(cmd)
+
+    def run(self):
+        self.run_for_a_duration(read_operation_event=FullScanEvent)
+
+
+class ReversedQueryThread(ReadOperationThread):
+    reversed_query_suffix = ' where pk = {} order by ck desc'
```

Comment:
This is not enough.
As I wrote you in the trello task, it should support other options like:
- ck limitations (e.g. ck > {}, ck >{} and ck < {}, ...).
- Limit (e.g .... order by ck desc LIMIT 100)
- Paging (e.g. retrive only X pages or retrive all).

Every query that you build, should have its matching "opposite" query (notice that it's not always simple because for example when you limit 100, you should retrieve all but take only the last 100).
This "opposite" query should be used to verify the result we got from the reversed query.
Make this comparison optional for now (controlled through the yaml params of run_revresed_query") because I don't know how well it behaves and if the results will be completely accurate.
(You'll need to test it).
In any case, if the results doesn't match, you should raise an ERROR event.

Reply:
fixed.